pkg/snowflake/db.go

```diff
@@ -222,3 +222,45 @@ func (db *DB) CreateSchemaIfNotExist(ctx context.Context, asset *pipeline.Asset)
 
 	return nil
 }
+
+func (db *DB) RecreateTableOnMaterializationTypeMismatch(ctx context.Context, asset *pipeline.Asset, errorMessage string, creationQuery *query.Query) error {
+	tableComponents := strings.Split(asset.Name, ".")
+	var schemaName string
+	var tableName string
+	switch len(tableComponents) {
+	case 2:
+		schemaName = strings.ToUpper(tableComponents[0])
+		tableName = strings.ToUpper(tableComponents[1])
+	case 3:
+		schemaName = strings.ToUpper(tableComponents[1])
+		tableName = strings.ToUpper(tableComponents[2])
+	default:
+		return nil
+	}
+	var materializationType string
+	switch {
+	case strings.Contains(errorMessage, "as TABLE"):
+
+		materializationType = "TABLE"
+
+	case strings.Contains(errorMessage, "as VIEW"):
+		materializationType = "VIEW"
+
+	default:
+		return nil
+	}
+
+	dropQuery := query.Query{
+		Query: fmt.Sprintf("DROP %s IF EXISTS %s.%s", materializationType, schemaName, tableName),
+	}
```

Comment:
why do we do this based on the error instead of comparing the type in the destination and the type in our query, without waiting for an error?